front/components/assistant/conversation/AttachmentCitation.tsx

```diff
@@ -0,0 +1,211 @@
+        <div className="flex gap-1 pt-1 text-sm">
+          <Icon visual={attachmentCitation.spaceIcon} />
+          <p>{attachmentCitation.spaceName}</p>
+        </div>
+        <div className="text-sm text-muted-foreground dark:text-muted-foreground-night">
+          {attachmentCitation.path}
+        </div>
+      </div>
+    );
+
+  return (
+    <Tooltip
+      trigger={
+        <Citation
+          className={className}
+          href={attachmentCitation.sourceUrl ?? undefined}
+          isLoading={
+            attachmentCitation.type === "file" && attachmentCitation.isUploading
+          }
+          action={
+            onRemove && (
+              <CitationClose
+                onClick={(e) => {
+                  e.stopPropagation();
+                  onRemove();
+                }}
+              />
+            )
+          }
+        >
+          {attachmentCitation.type === "file" && attachmentCitation.preview && (
+            <CitationImage imgSrc={attachmentCitation.preview} />
+          )}
+          <CitationIcons>{attachmentCitation.visual}</CitationIcons>
+          <CitationTitle className="truncate text-ellipsis">
+            {attachmentCitation.title}
+          </CitationTitle>
+          {attachmentCitation.type === "node" && (
+            <CitationDescription className="truncate text-ellipsis">
+              <div className="flex items-center gap-1">
+                <span>{attachmentCitation.spaceName}</span>
+              </div>
+            </CitationDescription>
+          )}
+        </Citation>
+      }
+      label={tooltipContent}
+    />
+  );
+}
+
+export function contentFragmentToAttachmentCitation(
+  contentFragment: ContentFragmentType
+): AttachmentCitation {
+  if (contentFragment.contentNodeData) {
+    const { provider, nodeType } = contentFragment.contentNodeData;
+    const logo = getConnectorProviderLogoWithFallback({ provider });
+
+    const visual =
+      provider && ["webcrawler", "folder"].includes(provider) ? (
+        <Icon visual={logo} size="sm" />
+      ) : (
+        <>
+          <Icon visual={logo} size="sm" />
+          <Icon
+            visual={
+              nodeType === "table"
+                ? TableIcon
+                : nodeType === "folder"
+                  ? FolderIcon
+                  : DocumentIcon
+            }
+            size="sm"
+          />
+        </>
+      );
+
+    return {
+      type: "node",
+      id: contentFragment.sId,
+      title: contentFragment.title,
+      sourceUrl: contentFragment.sourceUrl,
+      visual,
+      spaceName: contentFragment.contentNodeData.spaceName,
+    };
+  }
+
+  const isImageType = contentFragment.contentType.startsWith("image/");
+  return {
+    type: "file",
+    id: contentFragment.sId,
+    title: contentFragment.title,
+    sourceUrl: contentFragment.sourceUrl,
+    visual: <Icon visual={isImageType ? ImageIcon : DocumentIcon} size="sm" />,
```

Comment:
Image icon is not visible 🙃 (seems like it's behind the preview image?)

<img width="162" alt="Screenshot 2025-04-03 at 15 10 16" src="https://github.com/user-attachments/assets/4c167d09-171b-4b12-b171-a1223cf25e84" />

Reply:
Yeah definitely an existing bug 😓